Away mode temperature fix for generic thermostat

estevez-dev opened the pull request:
## Description:
Away mode temperature issue fix for generic_thermostat

**Related issue (if applicable):** fixes #17433 

## Example entry for `configuration.yaml` (if applicable):
```yaml
climate:
  - platform: generic_thermostat
    name: Child room heater
    heater: switch.40000500dc4f22ccb55e
    target_sensor: sensor.unknown_lumisensor_ht_0222819a_1_temperature
    min_temp: 10
    max_temp: 25
    ac_mode: false
    target_temp: 22
    cold_tolerance: 0.2
    hot_tolerance: 0
    initial_operation_mode: "off"
    away_temp: 10
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

Commit:
Resolves /home-assistant/home-assistant#17433

Away mode temperature issue fix for generic_thermostat

homeassistant commented:
Hi @estevez-dev, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#17641).

Once you do that we will be able to review and accept this pull request.

Thanks!

Commit:
Debug messages removed from generic_thermostat.py

Danielhiversen commented:
Could you add a test for this?

estevez-dev commented:
> Could you add a test for this?

Unfortunately I am complete zero in python. It would be very nice if someone more experienced will help me with it.

Commit:
Test for repeat away_mode set

Test for fix of generic thermostat issue when away_mode was set several times in a row.

estevez-dev commented:
Ok, I've added a test for setting away_mode twice. @Danielhiversen please let me know is it acceptable.

Danielhiversen reviewed (changes requested):
Thanks.
Just two small details

Commit:
Code style fix in generic_thermostat

estevez-dev commented on homeassistant/components/climate/generic_thermostat.py:
```diff
@@ -380,15 +380,17 @@ def is_away_mode_on(self):
-        self._target_temp = self._away_temp
-        await self._async_control_heating()
-        await self.async_update_ha_state()
+        if not self._is_away:
```
Done

estevez-dev commented:
Done

Commit:
Remove blank line in the end of generic_thermostat

Danielhiversen committed:
Fix style

Danielhiversen commented:
:thanks
